controllers/discovery/searchUser.js

```diff
@@ -9,7 +9,7 @@ const {sequelize} = require('../../databases/models');
  * @returns
  */
 const SearchUser = async (req, res) => {
-  const {q, limit = 50} = req.query;
+  const {q, limit = 50, allow_anon_dm} = req.query;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new parameter `allow_anon_dm` is not validated before use. It's important to validate user inputs to prevent potential security issues such as SQL injection attacks, and to ensure the application behaves correctly.
```diff
- const {q, limit = 50, allow_anon_dm} = req.query;
+ const {q, limit = 50, allow_anon_dm = false} = req.query;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->